src/QsCompiler/Compiler/QirDriverGeneration.cs

```diff
@@ -10,6 +13,8 @@ namespace Microsoft.Quantum.QsCompiler
 {
     public static class QirDriverGeneration
     {
+        private delegate string StringConversion(object? value);
```

Comment:
It might be worth making this a generic delegate:
```csharp
private delegate string StringConversion<T>(T? value);
```

That way the `GetBoolValueString`, `GetDoubleValueString`, and the other similar methods can be typed so you don't have to do a dynamic cast inside the functions. It is also more type-safe that way because if a type error occurs, the compiler will catch it.

Reply:
This doesn't actually work because you can't case these methods to a delegate with type arguments because they themselves do not have type arguments

Reply:
Also, `T?` is not supported in this context until C# 9